app/components/developer.py

```diff
@@ -34,10 +48,104 @@ async def status(self, interaction: dc.Interaction) -> None:
+    ) -> None:
+        # The client-side check with `default_permissions` isn't guaranteed to work.
+        if not self.bot.is_ghostty_mod(interaction.user):
+            await interaction.response.send_message(
+                "Only mods can use this command.", ephemeral=True
+            )
+            return
+
+        # If no extension is provided, reload all extensions
+        if extension:
+            if not self.bot.is_valid_extension(extension):
+                await interaction.response.send_message(
+                    f"{extension} is an invalid / unknown extension.", ephemeral=True
+                )
+                return
+            extensions = [extension]
+        else:
+            extensions = self.bot.get_component_extension_names()
+        reloaded_cogs: list[str] = []
+
+        await interaction.response.defer(thinking=True, ephemeral=True)
+        for cog in extensions:
+            await self.bot.unload_extension(cog)
+            await self.bot.load_extension(cog)
+            reloaded_cogs.append(cog)
+        await interaction.followup.send(f"Reloaded {reloaded_cogs}", ephemeral=True)
+
+    @dc.app_commands.command(description="Unload bot extension.")
+    @dc.app_commands.guild_only()
+    # Hide interaction from non-mods
+    @dc.app_commands.default_permissions(ban_members=True)
+    @dc.app_commands.autocomplete(extension=existing_extension_autocomplete)
+    async def unload(self, interaction: dc.Interaction, extension: str) -> None:
+        # The client-side check with `default_permissions` isn't guaranteed to work.
+        if not self.bot.is_ghostty_mod(interaction.user):
+            await interaction.response.send_message(
+                "Only mods can use this command.", ephemeral=True
+            )
+            return
+        if not self.bot.is_valid_extension(extension):
+            await interaction.response.send_message(
+                f"{extension} is an invalid / unknown extension.", ephemeral=True
+            )
+            return
+
+        await self.bot.unload_extension(extension)
+        await interaction.response.send_message(f"Unloaded {extension}", ephemeral=True)
+
+    @dc.app_commands.command(description="Load bot extension.")
+    @dc.app_commands.guild_only()
+    # Hide interaction from non-mods
+    @dc.app_commands.default_permissions(ban_members=True)
+    async def load(self, interaction: dc.Interaction, extension: str) -> None:
+        # The client-side check with `default_permissions` isn't guaranteed to work.
+        if not self.bot.is_ghostty_mod(interaction.user):
+            await interaction.response.send_message(
+                "Only mods can use this command.", ephemeral=True
+            )
+            return
+        if not self.bot.is_valid_extension(extension):
+            await interaction.response.send_message(
+                f"{extension} is an invalid / unknown extension.", ephemeral=True
```

Comment:
Ditto re unknown message.

```suggestion
                f"{extension} is an invalid/unknown extension.", ephemeral=True
```

```suggestion
                f"Extension `{extension}` does not exist or is invalid.", ephemeral=True
```